lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -440,95 +487,258 @@ def mock_run(args, **kwargs):
+                                            }
+                                        }
+                                    }
+                                },
+                                'subdir12':
+                                {
+                                    'details': <pbench.server.cache_manager.FileInfo object at 0x7fc4f1dc48d0>
+                                },
+                                'subdir11':
+                                {
+                                    'details': <pbench.server.cache_manager.FileInfo object at 0x7fc4f1d98850>
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            """
+
             m.setattr(Tarball, "__init__", TestCacheManager.MockTarball.__init__)
             m.setattr(Controller, "__init__", TestCacheManager.MockController.__init__)
-            m.setattr(Path, "glob", mock_glob)
-            m.setattr(os.path, "getmtime", lambda self: 1681814586.1138802)
-            m.setattr(Path, "is_dir", lambda self: True)
-            m.setattr(Path, "is_file", lambda self: True)
             tb = Tarball(tar, Controller(Path("/mock/archive"), cache, None))
-            tb.unpack()
-            assert tb.cachemap
-            assert tb.cachemap["A"]["dir1"]["details"].location == Path("/mock/A/dir1")
-            assert (
-                tb.cachemap["A"]["dir1"]["dir2"]["details"].timestamp
-                == "2023-04-18 16:13:06"
-            )
-            assert call == ["tar", "find"]
-            assert tb.unpacked == cache / "ABC" / tb.name
+            tar_dir = TestCacheManager.MockTarball.create_tarball("tar_name")
+            tb.cache_map(tar_dir)
+            assert tb.cachemap["tar_name"]["details"].type == "directory"
+
+            # test get_info with a file path
+            file_info = tb.get_info(Path("tar_name/subdir1/f1.txt"))
+            expected_info = {
+                "location": Path("/tmp/tar_name/subdir1/f1.txt"),
+                "name": "f1.txt",
+                "size": "1024",
+                "type": "file",
+            }
+            assert file_info == expected_info
+
+            # test get_info with a directory path
+            dir_info = tb.get_info(Path("tar_name/subdir1"))
+            expected_info = {
+                "directory": ["subdir13", "subdir12", "subdir11"],
+                "file": ["f1.txt"],
+            }
+            assert dir_info == expected_info
+
+            # test get_info with an empty directory path
+            dir_info = tb.get_info(Path("tar_name/subdir1/subdir11"))
+            expected_info = {
+                "location": Path("/tmp/tar_name/subdir1/subdir11"),
+                "name": "subdir11",
+                "type": "directory",
+            }
+            assert dir_info == expected_info
```

Comment:
Why does the info at line 715 _not_ expect a `name` and `type`?

And, correspondingly, it seems odd for the info at line 723 not to indicate any subdirectories or files except by the fact that the keys are missing -- given that it is normal for directories to contain entries, it seems like the keys should be present even if their values are empty lists.

We should meet some level of consistency for the caller.